crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -76,12 +111,88 @@ impl Settlement {
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    #[error(transparent)]
-    Solution(#[from] solution::Error),
+    #[error("failed communication with the database: {0}")]
+    BadCommunication(sqlx::Error),
+    #[error("failed to prepare the data fetched from database for domain: {0}")]
+    BadPersistenceData(BadPersistenceData),
     #[error("settlement refers to an auction from a different environment")]
     WrongEnvironment,
-    #[error("connection to the persistence layer failed: {0}")]
-    PersistenceConnection(#[from] infra::persistence::Error),
     #[error(transparent)]
-    Auction(#[from] infra::persistence::error::Auction),
+    BuildingSolution(#[from] solution::Error),
+    #[error(transparent)]
+    BuildingScore(#[from] solution::error::Score),
```

Comment:
Isn't building score also a data inconsistency (can only happen if score is 0, a price is missing or the math overflows)?

Reply:
I wanted to differentiate between errors from infra versus errors from calculations done inside Settlement constructor.

Let's re-evaluate if this is needed in the 4/4 PR.